src/ReverseProxy/Forwarder/StreamCopyHttpContent.cs

```diff
@@ -113,56 +149,89 @@ public StreamCopyHttpContent(Stream source, bool autoFlushHttpClientOutgoingStre
         /// we have full control over pumping bytes to the target stream for all protocols
         /// (except Web Sockets, which is handled separately).
         /// </remarks>
-        // Note we do not need to implement the SerializeToStreamAsync(Stream stream, TransportContext context, CancellationToken token) overload
-        // because the token it provides is the short request token, not the long body token passed from the constructor. Using the short token
-        // would break bidirectional streaming scenarios.
-        protected override async Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+        protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+        {
+            return SerializeToStreamAsync(stream, context, CancellationToken.None);
+        }
+
+#if NET
+        protected override
+#else
+        private
+#endif
+            async Task SerializeToStreamAsync(Stream stream, TransportContext? context, CancellationToken cancellationToken)
         {
             if (Interlocked.Exchange(ref _started, 1) == 1)
             {
                 throw new InvalidOperationException("Stream was already consumed.");
             }
 
-            if (_autoFlushHttpClientOutgoingStream)
+            // The cancellationToken that is passed to this method is:
+            // On HTTP/1.1: Linked HttpContext.RequestAborted + Request Timeout
+            // On HTTP/2.0: SocketsHttpHandler error / the server wants us to stop sending content / H2 connection closed
+            // The _contentCancellation is HttpContext.RequestAborted, so on HTTP/1.1 we don't have to combine the tokens
+            CancellationTokenSource? linkedCts = null;
+
+#if NET
+            if (_requestCancellation != cancellationToken)
             {
-                // HttpClient's machinery keeps an internal buffer that doesn't get flushed to the socket on every write.
-                // Some protocols (e.g. gRPC) may rely on specific bytes being sent, and HttpClient's buffering would prevent it.
-                // AutoFlushingStream delegates to the provided stream, adding calls to FlushAsync on every WriteAsync.
-                // Note that HttpClient does NOT call Flush on the underlying socket, so the perf impact of this is expected to be small.
-                // This statement is based on current knowledge as of .NET Core 3.1.201.
-                stream = new AutoFlushingStream(stream);
+                linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_contentCancellation, cancellationToken);
```

Comment:
No point in creating this if it can't be cancelled.
```suggestion
                Debug.Assert(cancellationToken.CanBeCanceled);
                linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_contentCancellation, cancellationToken);
```